File: Python_Codes/12. Enums & Annotations/BuiltInAnnotationsMain.py
import warnings
import functools

# Custom decorator to mark deprecated methods
def deprecated(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        warnings.warn(
            f"{func.__name__} is deprecated and may be removed in future versions.",
            category=DeprecationWarning,
            stacklevel=2
        )
        return func(*args, **kwargs)
    return wrapper

class BuiltInAnnotationsExample:
    
    # Simulating @SuppressWarnings by ignoring warnings manually
    def suppress_warning_example(self):
        warnings.filterwarnings("ignore", category=DeprecationWarning)
        raw_list = []  # Python doesn't have raw types, but we can simulate unchecked behavior
        raw_list.append("This is a raw type example!")
        print(raw_list[0])
        warnings.resetwarnings()

    # Simulating @Deprecated
    @deprecated
    def deprecated_method(self):
        print("This method is deprecated and may be removed in future versions.")

    # Simulating @Override by overriding __str__
    def __str__(self):
        return "BuiltInAnnotationsExample class demonstrating built-in annotations!"

# Main execution
if __name__ == "__main__":
    example = BuiltInAnnotationsExample()

    # Calling the deprecated method
    example.deprecated_method()

    # Suppressing warnings example
    example.suppress_warning_example()

    # Using the overridden __str__ method
    print(example)

File: Java_Codes/12. Enums & Annotations/BuiltInAnnotationsMain.java
// Demonstrating built-in annotations
class BuiltInAnnotationsExample {

    // SuppressWarnings annotation to ignore compiler warnings
    @SuppressWarnings("unchecked")
    public void suppressWarningExample() {
        // Example usage of raw types (unchecked warning suppressed)
        @SuppressWarnings("rawtypes")
        java.util.ArrayList rawList = new java.util.ArrayList();
        rawList.add("This is a raw type example!");
        System.out.println(rawList.get(0));
    }

    // Deprecated annotation to mark deprecated methods
    @Deprecated
    public void deprecatedMethod() {
        System.out.println("This method is deprecated and may be removed in future versions.");
    }

    // Override annotation to ensure method overrides
    @Override
    public String toString() {
        return "BuiltInAnnotationsExample class demonstrating built-in annotations!";
    }
}

// Main class to test built-in annotations
public class BuiltInAnnotationsMain {
    public static void main(String[] args) {
        BuiltInAnnotationsExample example = new BuiltInAnnotationsExample();

        // Calling the deprecated method
        example.deprecatedMethod();

        // Suppressing warnings example
        example.suppressWarningExample();

        // Using the overridden toString method
        System.out.println(example.toString());
    }
}